### PR TITLE
The following permissions were added:

### PR DESCRIPTION
**Pull request by zefciu**
* Superusers can do all
* Users can only edit domains/records they own
* Users can't subdomain domains they don't own
* Users can't create records in domains they don't own

### CODE REVIEW

File: powerdns/admin.py [hunk 1]
```diff
@@ -12,10 +12,14 @@
     SuperMaster,
 )
 
+import rules
```
Comment (andrzej-jankowski) [inline powerdns/admin.py]: This should be over the "from ..." imports.

File: powerdns/admin.py [hunk 2]
```diff
@@ -52,12 +56,18 @@ def queryset(self, request, queryset):
...
 class RecordAdminForm(ModelForm):
+
     def clean_type(self):
         type = self.cleaned_data['type']
         if not type:
             raise ValidationError(_("Record type is required"))
         return type
 
+    def clean_domain(self):
+        validator = PermissionValidator('powerdns.change_domain')
```
Comment (andrzej-jankowski) [inline powerdns/admin.py]: Why ``user`` is not optional param in ``PermissionValidator.__init__``?

Comment (andrzej-jankowski) [inline powerdns/admin.py]: How can I disable perms mechanizm? Maybe I don't need it...

Comment (zefciu) [inline powerdns/admin.py]: ad 1. In most cases the user is set in 'set_context' method of UserBasedValidator
ad 2. You can go with superusers only

**Comment by andrzej-jankowski** [comment]
Missing documentation. The rest is OK.

**Pull request merged**